crates/hyperqueue/src/server/client/autoalloc.rs

```diff
@@ -238,6 +242,206 @@ fn create_queue(
+    if let Some(q_type) = get_arg_by_key(String::from("-q"), args) {
+        let queue_type = q_type.replace("-q", "");
+        let queue_info = Command::new("qstat")
+            .arg("-Q")
+            .arg(&queue_type)
+            .arg("-f")
+            .arg("-F")
+            .arg("json")
+            .output()?;
+
+        let output = String::from_utf8_lossy(&queue_info.stdout).into_owned();
+        let json: serde_json::Value =
+            serde_json::from_str(&*output).expect("JSON was not well-formatted");
+        let queue = json
+            .get("Queue")
+            .and_then(|x| x.get(queue_type))
+            .context("can't find queue info")?;
+
+        let (walltime, backlog, workers) = match q_type.as_str() {
+            "qfat" => (
+                "resources_max/walltime",
+                "max_queued",
+                "resources_available/nodes",
+            ),
+            "qviz" => (
+                "resources_max/walltime",
+                "max_queued",
+                "resources_available/nodes",
+            ),
+            "qprod" => (
+                "resources_max/walltime",
+                "max_queued",
+                "resources_available/nodes",
+            ),
+            "qexp" => (
+                "resources_max/walltime",
+                "max_queued",
+                "resources_max/nodect",
+            ),
+            "qfree" => (
+                "resources_max/walltime",
+                "max_queued",
+                "resources_max/nodect",
+            ),
+            "qlong" => (
+                "resources_max/walltime",
+                "max_queued",
+                "resources_max/nodect",
+            ),
+            "qnvidia" => (
+                "resources_max/walltime",
+                "max_queued",
+                "resources_available/nodes",
+            ),
+            "qnvidia_test" => (
+                "resources_max/walltime",
+                "max_queued",
+                "resources_available/nodes",
+            ),
+            "qdgx" => (
+                "resources_max/walltime",
+                "max_queued",
+                "resources_available/nodes",
+            ),
+            _ => return Ok(None),
+        };
+
+        let (walltime, backlog, workers) = (
+            match get_value_by_key_json(queue, String::from(walltime)) {
+                Ok(value) => {
+                    let walltime =
+                        NaiveTime::from_str(value.as_str().context("can't convert str")?)?;
+                    Some(walltime.hour() * 3600 + walltime.minute() * 60 + walltime.second())
+                }
+                Err(err) => {
+                    log::warn!("Can't find key {walltime}: {err}");
+                    None
+                }
+            },
+            match get_value_by_key_json(queue, String::from(backlog)) {
+                Ok(value) => Some(
+                    value
+                        .to_string()
```

Comment:
Why is this `is_numeric` filter here? What data is returned by PBS here?

Reply:
Backlog has specific format in our qstat: "max_queued":"[u:PBS_GENERIC=5]"